### PR TITLE
Go-to setpoint type

### PR DESCRIPTION
**Pull request by tstastny**
Adds a "Go-to" setpoint type which streams a position and (optional) heading setpoint, meant to be handled by a position and heading smoother on the flight controller. Main idea: dump in whatever position and heading you want, and they will be smoothly followed by the autopilot. Additional fields for velocity and heading rate limits are included.

An example mode using the Go-to setpoint type is included:
https://github.com/Auterion/px4-ros2-interface-lib/assets/8026163/76795ebc-13b9-4637-b7f7-867e773047fa

Depends on: https://github.com/PX4/PX4-Autopilot/pull/22375

Docs: https://github.com/PX4/PX4-user_guide/pull/2935

### CODE REVIEW

File: examples/cpp/modes/goto/include/mode.hpp [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+{
+public:
+  explicit FlightModeTest(rclcpp::Node & node)
+  : ModeBase(node, kName)
+  {
+    _goto_setpoint = std::make_shared<px4_ros2::GotoSetpointType>(*this);
+
+    _vehicle_local_position = std::make_shared<px4_ros2::OdometryLocalPosition>(*this);
+
+    _vehicle_attitude_sub = node.create_subscription<px4_msgs::msg::VehicleAttitude>(
```
Comment (bkueng) [inline examples/cpp/modes/goto/include/mode.hpp:31]: We should add this to the odometry

File: examples/cpp/modes/goto/include/util.hpp [hunk 2]
```diff
@@ -0,0 +1,43 @@
+/****************************************************************************
+ * Copyright (c) 2023 PX4 Development Team.
+ * SPDX-License-Identifier: BSD-3-Clause
+ ****************************************************************************/
+#pragma once
+
+#include <Eigen/Eigen>
+#include <cmath>
+
+static inline Eigen::Vector3f quaternionToEuler(const Eigen::Quaternionf & q)
```
Comment (bkueng) [inline examples/cpp/modes/goto/include/util.hpp:10]: These should be made generally available under some `math/conversion` dir

File: examples/cpp/modes/goto/include/mode.hpp [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+    }
+  }
+
+private:
+  static constexpr float kTriangleHeight = 20.F; // [m]
+  static constexpr float kTriangleWidth = 30.F; // [m]
+
+  enum class State
+  {
+    kSettlingAtStart = 0,
```
Comment (bkueng) [inline examples/cpp/modes/goto/include/mode.hpp:150]: enum's don't use the constants naming
```suggestion
    SettlingAtStart = 0,
```

**Review by bkueng — COMMENTED** [review]
Looks good, we can also make the attitude & conversions generally available later on.

File: examples/cpp/modes/goto/include/mode.hpp (hunk 3 above)
Comment (tstastny) [reply examples/cpp/modes/goto/include/mode.hpp]: done

File: examples/cpp/modes/goto/include/util.hpp (hunk 2 above)
Comment (tstastny) [reply examples/cpp/modes/goto/include/util.hpp:12]: yeah i agree .. just didnt do it yet for the sake of getting the example out quick - added ugly note to shame myself until we consider where to better move these utils

File: examples/cpp/modes/goto/include/mode.hpp (hunk 1 above)
Comment (tstastny) [reply examples/cpp/modes/goto/include/mode.hpp:31]: agree - perhaps a follow up pr..

**Comment by tstastny** [comment]
@bkueng made quick minimal changes - but the rest i think we just do in a follow up to enable folks quickly on a working example for goto, ok for you?

**Pull request merged**